llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -90,17 +82,21 @@ def data(self):
 
         Use `get` and `set` instead.
         """
+        msg = "`data` is deprecated, please use the `get` and `set` method to store data into the Context."
+        warnings.warn(msg, DeprecationWarning)
         return self._globals
 
     @property
     def lock(self) -> asyncio.Lock:
         """Returns a mutex to lock the Context."""
-        return self._parent._lock if self._parent else self._lock
+        return self._lock
 
     @property
-    def session(self) -> "WorkflowSession":
-        """Returns a mutex to lock the Context."""
-        return self._parent._session if self._parent else self._session
+    def session(self) -> "Context":
```

Comment:
Great! Lets update any docs/examples that reference this either in this PR or a followup 👍🏻

Reply:
If you think the changelog is manageable I can do it in this PR

Reply:
Yea lets just do it in this PR, changeling should be fine 👍🏻